crates/uv/src/commands/help.rs

```diff
@@ -49,3 +68,30 @@ fn find_command<'a>(
     let subcommand = cmd.find_subcommand(next).ok_or((query, cmd))?;
     find_command(&query[1..], subcommand)
 }
+
+/// Write the contents of documentation to disk and spawn the given command to
+/// display it.
+fn write_and_spawn(name: &str, contents: impl Display, command: &str, args: &[&str]) -> Result<()> {
+    // Note this implementation is based on Cargo's `help` command which displays a `man` page.
+    // See https://github.com/rust-lang/cargo/blob/e98f702d83d3075b2232b4502114f177826fe06d/src/bin/cargo/commands/help.rs
+    use std::io::Write;
+
+    let prefix = format!("uv-{name}.");
+    let mut tmp = tempfile::Builder::new().prefix(&prefix).tempfile()?;
+    let f = tmp.as_file_mut();
+    write!(f, "{contents}")?;
+    f.flush()?;
+
+    let path = tmp.path();
+    // Use a path relative to the temp directory so that it can work on
+    // cygwin/msys systems which don't handle windows-style paths.
+    let mut relative_name = std::ffi::OsString::from("./");
+    relative_name.push(path.file_name().unwrap());
+    let mut cmd = std::process::Command::new(command)
```

Comment:
Curious, is it possible to use stdin instead of relying on a temporary file for this?

Reply:
Quite possibly. I copied Cargo's implementation nearly wholesale here — I'm not sure why it's implemented that way.

Reply:
It seems that Cargo uses a temporary file because it supports displaying content using `man`, which does not handle pipeing from stdio. 🤔 
https://github.com/rust-lang/cargo/commit/0e26eae5c1c3928e65921c40a667248338a2a92b#diff-5c9d825bb8a2ee825d064356b2f339ec5fbcc699ee617b484a703e03907884e2R54-R59

Reply:
Makes sense to me. I'll consider switching here, it should be easy. I wondered about `man` pages but decided not to bite that off yet.

Reply:
Switched to a pipe